docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+* Displaying messages in reverse order
+* Getting all messages ever sent
+
+Based on the parameters you use in your fetch command, you can easily satisfy any of the use cases above.
+
+A significant point to remember - **messages must be unique in sorted sets**! This means when saving the message to the sorted set, you should include some sort of message identifier or timestamp in your JSON.
+
+#### List
+
+A [list](../../cache/develop/basics/datatypes#lists) is a collection data type that stores an ordered array of elements. You can push elements to either the front or back of a list and you can add multiple entries at one time. A list is a great data type for instant messaging because it stores messages in the order they were sent. When users join an instant messaging session, you can quickly fetch the entire chat history with a single [listFetch API call](../../cache/develop/api-reference/list-collections#listfetch).
+
+Elements in a list do not need to be unique. You can push the same message into a list over and over again and it will be added every time. Lists are a good choice for instant messaging for the following use cases:
+
+* Paging messages N at a time
+* Adding multiple messages at once
+* Maintaining a chat history of fixed length, i.e. the last 100 messages
+
+When using a list, you have more array manipulation functions than with a sorted set, allowing you to automatically truncate message history if it gets too long or remove specific elements for things like content moderation.
+
+### Delivery
+
+Technically, message storage is not a necessary component for instant messaging. In theory you could have a messaging system that only showed the messages that were sent while a user is connected. Messages could be stored in memory in the client and you'd be done. That said, message delivery is a mandatory part of instant messaging. Getting the message from the sender to all recipients with minimal latency is the core of this pattern. To do this, we use [Momento Topics](https://www.gomomento.com/platform/topics).
+
+Momento Topics is a low-latency pub/sub service that connects clients together, clients to servers, servers to clients, and servers to servers. Think WebSockets but *way better*. When you publish a message to a topic, Momento will deliver it to all subscribers of the topic, providing fan-out capability or 1 to 1, depending on your use case.
+
+For instant messaging, the sender will *publish* a message to a topic. The recipients of the message, be it one or many depending on your use case, will be *subscribers*. Momento Topics will deliver the published message to the subscribers in real-time, providing for the instant messaging experience we're looking for!
+
+:::tip
+Topics are dynamic and are not defined resources in your infrastructure. The recommended way to build instant messaging is to *use the chat room or session identifier as the topic name*. This provides flexible and dedicated topics for chat use cases.
```

Comment:
```suggestion
Momento Topics are dynamic and are not defined resources in your infrastructure. The recommended way to build instant messaging is to *use the chat room or session identifier as the topic name*. This provides flexible and dedicated topics for chat use cases.
```

Reply:
I'm not a fan of this suggestion because it mixes the product name "Momento Topics" with "topics" the...thing in the product. I will reword for less confusion though